### PR TITLE
added property "allow_path_on_hierarchy" to JAX-RS plugin

### PR DESCRIPTION
**Pull request by Theoderich**
to optionally allow it to find jax-rs transactions on classes where only the interface or superlass is annotated with @Path

### CODE REVIEW

**Comment by felixbarny** [comment]
Hi and thanks for this PR 🙂 
Have you measured how much slower the startup is for your application?

File: apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Configuration provider for the apm jax-rs plugin
+ */
+public class JaxRsConfiguration extends ConfigurationOptionProvider {
+    private static final String JAXRS_CATEGORY = "jaxrs";
+
+    private final ConfigurationOption<Boolean> allowPathOnHierarchy = ConfigurationOption.booleanOption()
+        .key("allow_path_on_hierarchy")
+        .configurationCategory(JAXRS_CATEGORY)
+        .tags("performance")
+        .description("If set to `true`,\n" +
```
Comment (felixbarny) [inline apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java]: Seems like the description is out-of sync with `configuration.asciidoc`. You can execute the test `ConfigurationExporterTest`, or just `mvn clean test` to re-generate `configuration.asciidoc`.

Comment (Theoderich) [reply apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java]: Oh, I wasn't aware that the asciidoc is auto generated ^^ Will fix

**Comment by Theoderich** [comment]
I made some measurements with our application (150k lines of code wildfly 15)
The application takes about 40s to start without "allow_path_on_hierarchy".
Enabling the "allow_path_on_hierarchy" flag adds 1-2 seconds startup time.

File: apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.stagemonitor.configuration.ConfigurationOptionProvider;
+
+/**
+ * Configuration provider for the apm jax-rs plugin
+ */
+public class JaxRsConfiguration extends ConfigurationOptionProvider {
+    private static final String JAXRS_CATEGORY = "JAX-RS";
+
+    private final ConfigurationOption<Boolean> allowPathOnHierarchy = ConfigurationOption.booleanOption()
+        .key("allow_path_on_hierarchy")
```
Comment (felixbarny) [inline apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java]: Just from looking at the configuration name it's not immediately apparent that it's about JAX-RS annotations.
I suggest renaming to `enable_jaxrs_annotation_inheritance`

**Comment by felixbarny** [comment]
Thanks for measuring ❤️ 
Did you also measure the startup time without the agent?

**Comment by Theoderich** [comment]
I have made all the suggested changes:
- property renamed
- default set to true
- refactored tests

Regarding the startup time with/without the agent, please see the following table

Without javaagent| javaagent without @Path inheritance | javaagent with @Path inheritance
------------ | -------------|-------------
~36.5s |  ~39s |  ~41s

Regarding the CLA:
I've individually signed it and 30 Minutes ago my company has signed it. No idea why the check fails

File: apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * Configuration provider for the apm jax-rs plugin
+ */
+public class JaxRsConfiguration extends ConfigurationOptionProvider {
+    private static final String JAXRS_CATEGORY = "JAX-RS";
+
+    private final ConfigurationOption<Boolean> enableJaxrsAnnotationInheritance = ConfigurationOption.booleanOption()
+        .key("enable_jaxrs_annotation_inheritance")
+        .configurationCategory(JAXRS_CATEGORY)
+        .tags("performance")
+        .description("According to JAX-RS 2.0 spec (section 3.6 Annotation Inheritance) inheritance of class or interface annotations is not supported.\n" +
```
Comment (felixbarny) [inline apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsConfiguration.java]: Adjust the description based on the fact that the default is true now. I also think we don't need to cite the spec here.

**Comment by eyalkoren** [comment]
@Theoderich I am terribly sorry, but there was a problem with our CLA signing and verification process. It seems impossible to recover your signature and apply it, so I would kindly ask you to sign it again.

**Comment by Theoderich** [comment]
> 
> 
> @Theoderich I am terribly sorry, but there was a problem with our CLA signing and verification process. It seems impossible to recover your signature and apply it, so I would kindly ask you to sign it again.

OK, signed it again just now

**Comment by Theoderich** [comment]
Now the check is passing :)

**Comment by eyalkoren** [comment]
@Theoderich That's great, thanks! 
@felixbarny anything else you want to comment/change on this one?

**Pull request merged**